### PR TITLE
Migrate from CardDAV sharing to more general DAV sharing

### PR DESCRIPTION
**Pull request by DeepDiver1975**
This PR basically moves files around and includes abstractions for a more general dav sharing approach.

Real caldav sharing will be handled in a follow up pr.

Please review - THX

### CODE REVIEW

File: apps/dav/lib/carddav/addressbook.php [hunk 1]
```diff
@@ -82,14 +82,14 @@ function getACL() {
 		}
 
 		// add the current user
-		if (isset($this->addressBookInfo['{' . \OCA\DAV\CardDAV\Sharing\Plugin::NS_OWNCLOUD . '}owner-principal'])) {
-			$owner = $this->addressBookInfo['{' . \OCA\DAV\CardDAV\Sharing\Plugin::NS_OWNCLOUD . '}owner-principal'];
+		if (isset($this->addressBookInfo['{' . \OCA\DAV\DAV\Sharing\Plugin::NS_OWNCLOUD . '}owner-principal'])) {
+			$owner = $this->addressBookInfo['{' . \OCA\DAV\DAV\Sharing\Plugin::NS_OWNCLOUD . '}owner-principal'];
```
Comment (PVince81) [inline apps/dav/lib/carddav/addressbook.php]: DAVDAV ? :smile:

File: apps/dav/lib/dav/sharing/backend.php [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+ * You should have received a copy of the GNU Affero General Public License, version 3,
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>
+ *
+ */
+
+namespace OCA\DAV\DAV\Sharing;
+
+use OCP\IDBConnection;
+
+class Backend {
```
Comment (PVince81) [inline apps/dav/lib/dav/sharing/backend.php]: no interface needed here I guess ?

File: apps/dav/lib/dav/sharing/backend.php [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+			->from('dav_shares')
+			->where($query->expr()->eq('resourceid', $query->createNamedParameter($resourceId)))
+			->andWhere($query->expr()->eq('type', $query->createNamedParameter($this->resourceType)))
+			->execute();
+
+		$shares = [];
+		while($row = $result->fetch()) {
+			$shares[]= [
+				'href' => "principal:${row['principaluri']}",
+//				'commonName' => isset($p['{DAV:}displayname']) ? $p['{DAV:}displayname'] : '',
```
Comment (PVince81) [inline apps/dav/lib/dav/sharing/backend.php]: this wasn't commented out in the original file

**Comment by PVince81** [comment]
Is the new "Backend" class tested by existing tests, perhaps indirectly through the ones for the CarddavBackend one ?

File: apps/dav/lib/carddav/addressbook.php (hunk 1 above)
Comment (DeepDiver1975) [inline apps/dav/lib/carddav/addressbook.php]: I know - one for the app one for the sub folder .... :lollipop:

File: apps/dav/lib/dav/sharing/backend.php (hunk 2 above)
Comment (DeepDiver1975) [inline apps/dav/lib/dav/sharing/backend.php]: no - this is just a class to allow reuse of methods ...

File: apps/dav/lib/dav/sharing/backend.php (hunk 3 above)
Comment (DeepDiver1975) [inline apps/dav/lib/dav/sharing/backend.php]: yeah - this is the only reason why the principal backend is/was really needed in here - I want to rethink the structuring of the code - I'll move this to a followup pr - this is not critical for now

**Comment by MorrisJobke** [comment]
Unit tests are broken

**Comment by DeepDiver1975** [comment]
bloody rebase fuckup - this is one commit too much :imp:

**Comment by MorrisJobke** [comment]
Still broken :see_no_evil:

**Comment by DeepDiver1975** [comment]
finally unit tests pass - note to self: do it right, man!

**Comment by DeepDiver1975** [comment]
> Is the new "Backend" class tested by existing tests, perhaps indirectly through the ones for the CarddavBackend one ?

we have a coverage on this file by 75%

**Comment by DeepDiver1975** [comment]
@MorrisJobke @PVince81 mind taking another look? THX

**Comment by MorrisJobke** [comment]
:+1:

**Comment by LukasReschke** [comment]
:+1:

**Pull request merged**